addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/private_id_disclosure.js

```diff
@@ -0,0 +1,90 @@
+// This script scans websocket messages for private IP V4 addresses as well as Amazon EC2 private hostnames
+// Private IP V$ such as: 10.x.x.x, 172.x.x.x, 192.168.x.x
+// Amazon EC2 private hostname: such as ip-10-0-56-78
+
+// * This Passive Scan base on http passive scan plugin:
+// ** org.zaproxy.zap.extension.pscanrules.TestInfoPrivateAddressDisclosure
+
+// * Regex test: https://regex101.com/r/SztaPj/2/
+
+// Author: Manos Kirtas (manolis.kirt@gmail.com)
+
+OPCODE_TEXT = 0x1;
+RISK_LOW 	= 1;
+CONFIDENCE_MEDIUM = 2;
+
+REGULAR_IP_OCTET = "(25[0-5]|2[0-4][0-9]|1?[0-9]{1,2})";
+REGULAR_PORTS = "(:(0|[1-9]\\d{0,3}|[1-5]\\d{4}|6[0-4]\\d{3}|65([0-4]\\d{2}|5[0-2]\\d|53[0-5]))\\b)?";
+
+var patternPre = [];
+
+/** Pattern for private IP V4 addresses as well as Amazon EC2 private hostnames */
+patternPre.push(
+    "(",
+    "\\b10\\.(",
+    REGULAR_IP_OCTET,
+    "\\.){2}",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\b172\\.",
+    "(3[01]|2[0-9]|1[6-9])\\.",
+    REGULAR_IP_OCTET,
+    "\\.",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\b192\\.168\\.",
+    REGULAR_IP_OCTET,
+    "\\.",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    // find IPs from AWS hostnames such as "ip-10-2-3-200"
+    "\\bip-10-(",
+    REGULAR_IP_OCTET,
+    "-){2}",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\bip-172-",
+    "(3[01]|2[0-9]|1[6-9])-",
+    REGULAR_IP_OCTET,
+    "-",
+    REGULAR_IP_OCTET,
+    "\\b|",
+    "\\bip-192-168-",
+    REGULAR_IP_OCTET,
+    "-",
+    REGULAR_IP_OCTET,
+    "\\b)",
+    REGULAR_PORTS
+);
+
+function scan(helper,msg) {
+
+    var ipRegex = new RegExp(patternPre.join(""),"gim");
```

Comment:
This could be done after the if statement at 64. No point processing/assembling the pattern if it isn't going to be used.

Reply:
Fixed in 068de3dec